### ISSUE TITLE
ApiGatewayResolver dynamic path route misbehaves when "." (dot) character in dynamic part

### ISSUE THREAD

**Issue by walmsles** [awslabs/aws-lambda-powertools-python#564]
Have defined a dynamic API path for the ApiGatewayResolver that is required to contain a "." (dot) character.  When the resolver is called the route does not match and resolver returns a 404 Not Found error message to the caller even though it is actually valid and correct.

Would seem the "." (dot) char is affecting the regex matching in some way.  I have changed my API to use a query parameter instead for now (but would prefer the dynamic path since is consistent with another API on the same domain which has been built with a different library and is nodeJS lambda).

**What were you trying to accomplish?**
Trying to create an API consistent with already deployed code which has dynamic path attribute for a user name which results in a path similar to this: ```/myapi/action/firstname.lastname``` with the last component of "firstname.lastname" is the dynamic component to be matched for the route ```api.post('/myapi/action/<username>)```

## Expected Behavior
GIVEN my app.route is defined as follows:
```python
@app.route("/myapi/action/<username>")
def myapi_action(username):
    return {"username": username}
```

WITH an API gateway path set up as: ```/myapi/action/{username}```
and the API path called via API gateway is: ```/myapi/action/firstname.lastname```

THEN I would expect the function ```myapi_action``` to be called with ```username == 'firstname.lastname'```

## Current Behavior
In the above scenario the ApiGatewayResolver.resolve() method returns a **404 NOT FOUND** response instead of calling my Api Handler. 
If I alternatively call the API via API gateway like this: ```/myapi/action/firstname``` then the API method works as expected.

## Possible Solution
Seems RegEx matching si thrown by regexy-like characters in the ApiGateway PATH while the route is trying to be matched.

## Steps to Reproduce (for bugs)
Let me know if my scenario above is not clear enough, happy to setup a sample project f required.

## Environment

* **Powertools version used**: 1.18.0
* **Packaging format (Layers, PyPi)**: PyPi
* **AWS Lambda function runtime:** Python 3.8
* **Debugging logs**
**NOTE: Logs below do not match my more common made up scenario above but illustrates the bug within my in dev API**
> [How to enable debug mode](https://awslabs.github.io/aws-lambda-powertools-python/#debug-mode)**

```python


START RequestId: 9dd4a8ce-3a3b-4f15-8136-9b36d92135d9 Version: $LATEST
--
2021-07-26 14:56:42,603 aws_lambda_powertools.tracing.tracer [DEBUG] Verifying whether Tracing has been disabled
2021-07-26 14:56:42,769 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Adding route using rule /chatbot/vaccinations/<eid> and method GET
2021-07-26 14:56:42,770 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Registering method GET to Allow Methods in CORS
2021-07-26 14:56:42,770 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Adding route using rule /vaccinations and method GET
2021-07-26 14:56:42,770 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Registering method GET to Allow Methods in CORS
2021-07-26 14:56:42,770 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Adding route using rule /vaccinations/status and method GET
2021-07-26 14:56:42,770 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Registering method GET to Allow Methods in CORS
2021-07-26 14:56:42,770 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Adding route using rule /chatbot/vaccinations/<eid> and method POST
2021-07-26 14:56:42,771 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Registering method POST to Allow Methods in CORS
2021-07-26 14:56:42,771 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Adding route using rule /vaccinations and method POST
2021-07-26 14:56:42,771 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Registering method POST to Allow Methods in CORS
2021-07-26 14:56:42,771 aws_lambda_powertools.logging.logger [DEBUG] Decorator called with parameters
2021-07-26 14:56:42,774 aws_lambda_powertools.logging.logger [DEBUG] Event received
{     "level": "INFO",     "location": "decorate:345",     "message": {         "resource": "/chatbot/vaccinations/{eid}",         "path": "/chatbot/vaccinations/m.walmsley",         "httpMethod": "POST",         "headers": {             "Accept": "*/*",             "Accept-Encoding": "gzip, deflate, br",             "CloudFront-Forwarded-Proto": "https",             "CloudFront-Is-Desktop-Viewer": "true",             "CloudFront-Is-Mobile-Viewer": "false",             "CloudFront-Is-SmartTV-Viewer": "false",             "CloudFront-Is-Tablet-Viewer": "false",             "CloudFront-Viewer-Country": "AU",             "Content-Type": "application/json",             "Host": "5t8c7qo8w1.execute-api.ap-southeast-1.amazonaws.com",             "Postman-Token": "6d0fd377-3820-4528-93d8-c64b344a30c0",             "User-Agent": "PostmanRuntime/7.28.1",             "Via": "1.1 5ecaba390628f1a4732482438ad91be5.cloudfront.net (CloudFront)",             "X-Amz-Cf-Id": "QLQx1sd46Rr9OFIdR7PpKMIFSiQxpwHuMDQpnmtqPD9i85Pf4sK9VA==",             "X-Amzn-Trace-Id": "Root=1-60fecd29-75d41c595af4c8c06be13398",             "x-api-key": "CuHTX0KCR15kMeExcUfgG40VDReDSdrP6KexMepO",             "X-Forwarded-For": "124.170.115.3, 130.176.156.110",             "X-Forwarded-Port": "443",             "X-Forwarded-Proto": "https"         },         "multiValueHeaders": {             "Accept": [                 "*/*"             ],             "Accept-Encoding": [                 "gzip, deflate, br"             ],             "CloudFront-Forwarded-Proto": [                 "https"             ],             "CloudFront-Is-Desktop-Viewer": [                 "true"             ],             "CloudFront-Is-Mobile-Viewer": [                 "false"             ],             "CloudFront-Is-SmartTV-Viewer": [                 "false"             ],             "CloudFront-Is-Tablet-Viewer": [                 "false"             ],             "CloudFront-Viewer-Country": [                 "AU"             ],             "Content-Type": [                 "application/json"             ],             "Host": [                 "5t8c7qo8w1.execute-api.ap-southeast-1.amazonaws.com"             ],             "Postman-Token": [                 "6d0fd377-3820-4528-93d8-c64b344a30c0"             ],             "User-Agent": [                 "PostmanRuntime/7.28.1"             ],             "Via": [                 "1.1 5ecaba390628f1a4732482438ad91be5.cloudfront.net (CloudFront)"             ],             "X-Amz-Cf-Id": [                 "QLQx1sd46Rr9OFIdR7PpKMIFSiQxpwHuMDQpnmtqPD9i85Pf4sK9VA=="             ],             "X-Amzn-Trace-Id": [                 "Root=1-60fecd29-75d41c595af4c8c06be13398"             ],             "x-api-key": [                 "CuHTX0KCR15kMeExcUfgG40VDReDSdrP6KexMepO"             ],             "X-Forwarded-For": [                 "124.170.115.3, 130.176.156.110"             ],             "X-Forwarded-Port": [                 "443"             ],             "X-Forwarded-Proto": [                 "https"             ]         },         "queryStringParameters": null,         "multiValueQueryStringParameters": null,         "pathParameters": {             "eid": "m.walmsley"         },         "stageVariables": null,         "requestContext": {             "resourceId": "a5r4iw",             "resourcePath": "/chatbot/vaccinations/{eid}",             "httpMethod": "POST",             "extendedRequestId": "DFT-iEPQyQ0FsKw=",             "requestTime": "26/Jul/2021:14:56:41 +0000",             "path": "/dev/chatbot/vaccinations/m.walmsley",             "accountId": "674048636348",             "protocol": "HTTP/1.1",             "stage": "dev",             "domainPrefix": "5t8c7qo8w1",             "requestTimeEpoch": 1627311401678,             "requestId": "215f5510-c551-4eca-a525-c0f61d6515ec",             "identity": {                 "cognitoIdentityPoolId": null,                 "cognitoIdentityId": null,                 "apiKey": "CuHTX0KCR15kMeExcUfgG40VDReDSdrP6KexMepO",                 "principalOrgId": null,                 "cognitoAuthenticationType": null,                 "userArn": null,                 "apiKeyId": "y6qew1qdq6",                 "userAgent": "PostmanRuntime/7.28.1",                 "accountId": null,                 "caller": null,                 "sourceIp": "124.170.115.3",                 "accessKey": null,                 "cognitoAuthenticationProvider": null,                 "user": null             },             "domainName": "5t8c7qo8w1.execute-api.ap-southeast-1.amazonaws.com",             "apiId": "5t8c7qo8w1"         },         "body": "{\n    \"accept_data_privacy\": true,\n    \"need_vaccination_help\": false,\n    \"vaccination_option\": \"partially_vaccinated\"\n}",         "isBase64Encoded": false     },     "timestamp": "2021-07-26 14:56:42,775+0000",     "service": "service_undefined",     "cold_start": true,     "function_name": "acn-mybuddy-vaccinations-dev-vaccinations",     "function_memory_size": "1024",     "function_arn": "arn:aws:lambda:ap-southeast-1:674048636348:function:acn-mybuddy-vaccinations-dev-vaccinations",     "function_request_id": "9dd4a8ce-3a3b-4f15-8136-9b36d92135d9",     "correlation_id": "215f5510-c551-4eca-a525-c0f61d6515ec",     "xray_trace_id": "1-60fecd29-75d41c595af4c8c06be13398" }
2021-07-26 14:56:42,775 aws_lambda_powertools.tracing.tracer [DEBUG] Annotating cold start
2021-07-26 14:56:42,775 aws_lambda_powertools.tracing.tracer [DEBUG] Calling lambda handler
2021-07-26 14:56:42,775 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Converting event to API Gateway REST API contract
2021-07-26 14:56:42,775 aws_lambda_powertools.event_handler.api_gateway [DEBUG] No match found for path /chatbot/vaccinations/m.walmsley and method POST
2021-07-26 14:56:42,775 aws_lambda_powertools.event_handler.api_gateway [DEBUG] CORS is enabled, updating headers.
2021-07-26 14:56:42,775 aws_lambda_powertools.tracing.tracer [DEBUG] Received lambda handler response successfully
END RequestId: 9dd4a8ce-3a3b-4f15-8136-9b36d92135d9
REPORT RequestId: 9dd4a8ce-3a3b-4f15-8136-9b36d92135d9	Duration: 3.09 ms	Billed Duration: 4 ms	Memory Size: 1024 MB	Max Memory Used: 83 MB	Init Duration: 897.51 ms

```

**Log of call WITHOUT "." (dot) character**

```python

START RequestId: c99d12c6-9195-4f6a-8f79-9662e35b52b1 Version: $LATEST
--
2021-07-26 14:56:47,531 aws_lambda_powertools.logging.logger [DEBUG] Event received
{     "level": "INFO",     "location": "decorate:345",     "message": {         "resource": "/chatbot/vaccinations/{eid}",         "path": "/chatbot/vaccinations/walmsley",         "httpMethod": "POST",         "headers": {             "Accept": "*/*",             "Accept-Encoding": "gzip, deflate, br",             "CloudFront-Forwarded-Proto": "https",             "CloudFront-Is-Desktop-Viewer": "true",             "CloudFront-Is-Mobile-Viewer": "false",             "CloudFront-Is-SmartTV-Viewer": "false",             "CloudFront-Is-Tablet-Viewer": "false",             "CloudFront-Viewer-Country": "AU",             "Content-Type": "application/json",             "Host": "5t8c7qo8w1.execute-api.ap-southeast-1.amazonaws.com",             "Postman-Token": "71937eee-053b-4d76-b7e8-f9fd4de9d466",             "User-Agent": "PostmanRuntime/7.28.1",             "Via": "1.1 5ecaba390628f1a4732482438ad91be5.cloudfront.net (CloudFront)",             "X-Amz-Cf-Id": "kvBzQPEYGnG7URkjT3XoSNzFdeZOQGlT2ZvTsucK1nIGHNswwBurfw==",             "X-Amzn-Trace-Id": "Root=1-60fecd2f-0a961fff7c7704513cc78bb4",             "x-api-key": "CuHTX0KCR15kMeExcUfgG40VDReDSdrP6KexMepO",             "X-Forwarded-For": "124.170.115.3, 130.176.156.110",             "X-Forwarded-Port": "443",             "X-Forwarded-Proto": "https"         },         "multiValueHeaders": {             "Accept": [                 "*/*"             ],             "Accept-Encoding": [                 "gzip, deflate, br"             ],             "CloudFront-Forwarded-Proto": [                 "https"             ],             "CloudFront-Is-Desktop-Viewer": [                 "true"             ],             "CloudFront-Is-Mobile-Viewer": [                 "false"             ],             "CloudFront-Is-SmartTV-Viewer": [                 "false"             ],             "CloudFront-Is-Tablet-Viewer": [                 "false"             ],             "CloudFront-Viewer-Country": [                 "AU"             ],             "Content-Type": [                 "application/json"             ],             "Host": [                 "5t8c7qo8w1.execute-api.ap-southeast-1.amazonaws.com"             ],             "Postman-Token": [                 "71937eee-053b-4d76-b7e8-f9fd4de9d466"             ],             "User-Agent": [                 "PostmanRuntime/7.28.1"             ],             "Via": [                 "1.1 5ecaba390628f1a4732482438ad91be5.cloudfront.net (CloudFront)"             ],             "X-Amz-Cf-Id": [                 "kvBzQPEYGnG7URkjT3XoSNzFdeZOQGlT2ZvTsucK1nIGHNswwBurfw=="             ],             "X-Amzn-Trace-Id": [                 "Root=1-60fecd2f-0a961fff7c7704513cc78bb4"             ],             "x-api-key": [                 "CuHTX0KCR15kMeExcUfgG40VDReDSdrP6KexMepO"             ],             "X-Forwarded-For": [                 "124.170.115.3, 130.176.156.110"             ],             "X-Forwarded-Port": [                 "443"             ],             "X-Forwarded-Proto": [                 "https"             ]         },         "queryStringParameters": null,         "multiValueQueryStringParameters": null,         "pathParameters": {             "eid": "walmsley"         },         "stageVariables": null,         "requestContext": {             "resourceId": "a5r4iw",             "resourcePath": "/chatbot/vaccinations/{eid}",             "httpMethod": "POST",             "extendedRequestId": "DFT_dG-PyQ0FveQ=",             "requestTime": "26/Jul/2021:14:56:47 +0000",             "path": "/dev/chatbot/vaccinations/walmsley",             "accountId": "674048636348",             "protocol": "HTTP/1.1",             "stage": "dev",             "domainPrefix": "5t8c7qo8w1",             "requestTimeEpoch": 1627311407514,             "requestId": "a37ae37c-7f99-4c42-be7c-6dc72dabfcf5",             "identity": {                 "cognitoIdentityPoolId": null,                 "cognitoIdentityId": null,                 "apiKey": "CuHTX0KCR15kMeExcUfgG40VDReDSdrP6KexMepO",                 "principalOrgId": null,                 "cognitoAuthenticationType": null,                 "userArn": null,                 "apiKeyId": "y6qew1qdq6",                 "userAgent": "PostmanRuntime/7.28.1",                 "accountId": null,                 "caller": null,                 "sourceIp": "124.170.115.3",                 "accessKey": null,                 "cognitoAuthenticationProvider": null,                 "user": null             },             "domainName": "5t8c7qo8w1.execute-api.ap-southeast-1.amazonaws.com",             "apiId": "5t8c7qo8w1"         },         "body": "{\n    \"accept_data_privacy\": true,\n    \"need_vaccination_help\": false,\n    \"vaccination_option\": \"partially_vaccinated\"\n}",         "isBase64Encoded": false     },     "timestamp": "2021-07-26 14:56:47,531+0000",     "service": "service_undefined",     "cold_start": false,     "function_name": "acn-mybuddy-vaccinations-dev-vaccinations",     "function_memory_size": "1024",     "function_arn": "arn:aws:lambda:ap-southeast-1:674048636348:function:acn-mybuddy-vaccinations-dev-vaccinations",     "function_request_id": "c99d12c6-9195-4f6a-8f79-9662e35b52b1",     "correlation_id": "a37ae37c-7f99-4c42-be7c-6dc72dabfcf5",     "xray_trace_id": "1-60fecd2f-0a961fff7c7704513cc78bb4" }
2021-07-26 14:56:47,531 aws_lambda_powertools.tracing.tracer [DEBUG] Calling lambda handler
2021-07-26 14:56:47,531 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Converting event to API Gateway REST API contract
2021-07-26 14:56:47,532 aws_lambda_powertools.event_handler.api_gateway [DEBUG] Found a registered route. Calling function
2021-07-26 14:56:47,532 aws_lambda_powertools.tracing.tracer [DEBUG] Calling method: post_chatbot_vaccinations

```

**Comment by heitorlessa** [comment]
Hey @walmsles, thanks again for spotting another one.

I believe this was fixed on 1.18.1 since we added support for safe and unsafe chars in the route - could you try with the latest version? 

I'll check tomorrow morning, but as you're 8 hours ahead, you might find it sooner.

Thanks again!

**Comment by heitorlessa** [comment]
Added an additional functional test and it works on 1.18.1 as expected, not on 1.18.0 due to regex previously not taking into account other non-word chars -- [this fix is in 1.18.1](https://github.com/awslabs/aws-lambda-powertools-python/pull/556/files#diff-bdb4b43087f89fdf398381f3d88048f6182562ca67e8907259d5c56d658465c1R24)

Let me know if 1.18.1 behaves differently for you and I can dig deeper.

```python
def test_route_with_dot():
    # GIVEN
    app = ApiGatewayResolver()
    event = deepcopy(LOAD_GW_EVENT)
    expected_username = "firstname.lastname"

    # WHEN
    @app.get("/myapi/action/<username>")
    def get_account(username: str):
        assert username == expected_username

    # THEN
    event["resource"] = "/myapi/action/{username}"  # API GW route match
    event["path"] = f"/myapi/action/{expected_username}"  # API GW incoming request

    ret = app.resolve(event, None)
    assert ret["statusCode"] == 200

```

**Comment by walmsles** [comment]
Hi @heitorlessa - correct!  Fixed as part of 1.18.1 - sorry for not updating and checking first.  Thanks so much for all your work

**Comment by walmsles** [comment]
Hi @heitorlessa - correct!  Fixed as part of 1.18.1 - sorry for not updating and checking first.  Thanks so much for all your work

**Comment by heitorlessa** [comment]
Awesome to hear! No worries, that's why we're here for ;)
